ToDoList/module/list/ListViewController.swift

```diff
@@ -7,56 +7,137 @@
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        viewModel.numberOfRows
     }
     
-   
-    @IBAction func notesToggleButtonTapped(_ sender: UIBarButtonItem) {
-        if let notesVC = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: "NotesViewController")
-                                                                                            as? NotesViewController {
-            notesVC.delegate = self
-            navigationController?.pushViewController(notesVC, animated: true)
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: ListCollectionViewCell.identifier, for: indexPath) as? ListCollectionViewCell else {
+           
+            return UICollectionViewCell()
         }
+        
+        let text = viewModel.cellForRowAt(at: indexPath)
+        let key = UserDefaultsKey.itemState(indexPath: indexPath)
+        let isChecked = UserDefaultsClass.shared.get(forKey: key)
+        
+        cell.configure(with: text, isChecked: isChecked, indexPath: indexPath)
+        return cell
     }
 }
 
+extension ListViewController: UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+       
+        return viewModel.sizeForItemAt(indexPath: indexPath, collectionViewWidth: collectionView.bounds.width)
+    }
+}
+
+extension ListViewController: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        func collectionView(_ collectionView: UICollectionView, trailingSwipeActionsConfigurationForItemAt indexPath: IndexPath) -> UISwipeActionsConfiguration? {
+            let deleteAction = UIContextualAction(style: .destructive, title: "Delete") { [weak self] (action, view, completionHandler) in
+                        self?.deleteItem(at: indexPath)
+                        completionHandler(true)
+                    }
+            let configuration = UISwipeActionsConfiguration(actions: [deleteAction])
+            configuration.performsFirstActionWithFullSwipe = false
+            return UISwipeActionsConfiguration(actions: [deleteAction])
+        }
+    }
+  
+}
+
 extension ListViewController: NotesViewControllerDelegate {
   
     func saveText(_ text: String) {
-        viewModel.list.append(text)
-        tableView.reloadData()
-        
-        UserDefaults.standard.set(viewModel.list, forKey: "textSave")
+        viewModel.saveText(text)
+    }
+}
+
+extension ListViewController: ListViewProtocol {
+    func navigateToNotesViewController() {
+        if let notesVC = self.storyboard?.instantiateViewController(ofType: NotesViewController.self) {
```

Comment:
create notes view model in here and set delegate

Reply:
Updated